src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1278,10 +1278,10 @@ private async ValueTask<Stream> ConnectToTcpHostAsync(string host, int port, Htt
 
             var endPoint = new DnsEndPoint(host, port);
             Socket? socket = null;
-            try
+            // If a ConnectCallback was supplied, use that to establish the connection.
+            if (Settings._connectCallback != null)
             {
-                // If a ConnectCallback was supplied, use that to establish the connection.
-                if (Settings._connectCallback != null)
+                try
```

Comment:
The old try catch wrapped both the ConnectCallback case as well as the default case. This one only wraps the ConnectCallback case. Why the change?

Reply:
There is an equivalent catch block in `ConnectHelper.ConnectAsync`:

https://github.com/dotnet/runtime/blob/6cba345ff9bcf2b55d27b64f0d5e08398eb10cd7/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs#L62-L65

https://github.com/dotnet/runtime/blob/6cba345ff9bcf2b55d27b64f0d5e08398eb10cd7/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs#L239-L244